api/wasm/cpp/proxy_wasm_enums.h

```diff
@@ -15,15 +15,29 @@ enum class FilterDataStatus : EnumType {
   StopIterationAndWatermark = 2,
   StopIterationNoBuffer = 3
 };
-enum class StreamType : EnumType { Request = 0, Response = 1 };
 enum class MetadataType : EnumType {
   Request = 0,
   Response = 1,
   RequestRoute = 2,   // Immutable
   ResponseRoute = 3,  // Immutable
   Log = 4,            // Immutable
-  Node = 5            // Immutable
+  Node = 5,           // Immutable
+  Listener = 6,       // Immutable
+  Cluster = 7,        // Immutable
+  Expression = 8,     // The key is a string expression. Only proxy_getMetadata().
 };
+/*
+  Expression and their types:
+
+  request.protocol : string
+  response.protocol : string
+  request.destination_port : int32
```

Comment:
We should be careful not to define new names for things that are established in https://istio.io/docs/reference/config/policy-and-telemetry/attribute-vocabulary/ or https://cloud.google.com/iam/docs/conditions-overview. I'm happy to make another pass to fix the names in a follow-up

Reply:
We should just use consistent names now, instead of another PR.

Reply:
I agree that we should have consistent names when we upstream, but this is an experimental repository and Kuat is going to replace all this code very soon.  This is just to get Mandar the last "must have" interface.  That said, if the correct names are obvious I'd like to make the change in this PR.

Reply:
The current names should not be consisdered part of the external interface.  We have the accessors for that, e.g. getRequestDestinationPort().  I can also prefix the current names with "_experimental_" or "_internal_" which will give a clear indication these are not part of a stable exernal interface.